apps/dotcom/sync-worker/src/TLPostgresReplicator.ts

```diff
@@ -120,190 +116,70 @@ export class TLPostgresReplicator extends DurableObject<Environment> {
 			// TODO: set a timeout on the promise?
-			promise: 'promise' in this.state ? this.state.promise : promiseWithResolve(),
-			db: postgres(this.env.BOTCOM_POSTGRES_CONNECTION_STRING, {
-				types: {
-					bigint: {
-						from: [20], // PostgreSQL OID for BIGINT
-						parse: (value: string) => Number(value), // Convert string to number
-						to: 20,
-						serialize: (value: number) => String(value), // Convert number to string
-					},
-				},
-			}),
-			bootId: uniqueId(),
+			promise,
+			db: getPostgres(this.env),
+			sequenceId: uniqueId(),
 		}
-		/**
-		 * BOOTUP SEQUENCE
-		 * 1. Generate a unique boot id
-		 * 2. Subscribe to all changes
-		 * 3. Fetch all data and write the boot id in a transaction
-		 * 4. If we receive events before the boot id update comes through, ignore them
-		 * 5. If we receive events after the boot id comes through but before we've finished
-		 *    fetching all data, buffer them and apply them after we've finished fetching all data.
-		 *    (not sure this is possible, but just in case)
-		 * 6. Once we've finished fetching all data, apply any buffered events
-		 */
-
-		const myId = this.ctx.id.toString()
-		// if the bootId changes during the boot process, we should stop silently
-		const bootId = this.state.bootId
-
-		const promise = this.state.promise
-
-		const start = Date.now()
-		const result = await new Promise<'error' | 'success'>((resolve) => {
-			let didFetchInitialData = false
-			let didGetBootIdInSubscription = false
-			let subscription: postgres.SubscriptionHandle | undefined
-
-			const initialUpdateBuffer: Array<{
-				row: postgres.Row | null
-				info: postgres.ReplicationEvent
-			}> = []
-
-			const updateState = () => {
-				if (didFetchInitialData && didGetBootIdInSubscription && subscription) {
-					// we got everything, so we can set the state to connected and apply any buffered events
-					assert(this.state.type === 'connecting', 'state should be connecting')
-					this.state = {
-						type: 'connected',
-						db: this.state.db,
-						bootId: this.state.bootId,
-						subscription,
-					}
-					this.debug('num updates pending after pg sync:', initialUpdateBuffer.length)
-					if (initialUpdateBuffer.length) {
-						while (initialUpdateBuffer.length) {
-							const { row, info } = initialUpdateBuffer.shift()!
-							this.handleEvent(row, info)
-						}
-					}
-					resolve('success')
+		const subscription = await this.state.db.subscribe(
+			'*',
+			this.handleEvent.bind(this),
+			() => {
+				// re-register all active users to get their latest guest info
+				const users = this.sql.exec('SELECT id FROM active_user').toArray()
+				for (const user of users) {
+					const sequenceId = this.state.sequenceId
+					this.queue.push(async () => {
+						if (this.state.sequenceId !== sequenceId) return
+						this.messageUser(user.id as string, { type: 'force_reboot', sequenceId })
+						await sleep(10)
```

Comment:
Why do we need this wait?

Reply:
to ease the pressure on the db during reboots. We might have a thousand or more actively connected users, and when the replicator reboots each of those users will reboot, and each user reboot triggers two sql queries (one in user DO to fetch their initial data, and one in replicator DO to fetch guest file subscriptions). 

I'd be worried about failures if kicking off a couple of thousand PG queries at exactly the same time. Staggering them individually by 10ms is definitely overkill, but I'll err one the side of caution for now and stress test this with batches of varying size in staging.